### PR TITLE
Multiarch docker containers

### LINKED ISSUE #4900: Multi-Arch docker images
It would be great if the automatic docker build process could be expanded for multi-arch images, something like this:
https://www.docker.com/blog/multi-arch-images/

This way people with ARM machines could use the latest client versions and help testing the bleeding edge.

### PR DESCRIPTION
**Pull request by prestonvanloon**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

This uses OCI manifest format (previously docker v2).

Features:
- [x] Images are now available in [OCI format](https://www.docker.com/blog/demystifying-open-container-initiative-oci-specifications/)
- [x] Images include /bin/bash
- [x] /bin/bash works on the images
- [x] Images are available for amd64 and arm64 as a multiarch index.
- [ ] All previous image uploads are supported (Will be completed in follow up PR)

**Which issues(s) does this PR fix?**

Fixes #4900

**Other notes for review**

~Depends on #12135, review and merge that first for clean diff.~

A follow up PR will provide the following changes:

- Remove obsolete rules_docker targets 
- Remove support for debug images
- Remove support for alpine images (These have been broken for a long time https://github.com/prysmaticlabs/prysm/issues/9320).

### CODE REVIEW

**Comment by batphonghan** [comment]
Hey @prestonvanloon . This's awesome work. Would like to ask when this PR can be merge and consider ready for production. I would like to use this for running in my arm machine. Thank you.

**Comment by batphonghan** [comment]
@prestonvanloon  Thank you for your quick response, wishing that you're fully recover now. Keep up the great work.

**Comment by prestonvanloon** [comment]
Blocked again in CI by https://github.com/bazel-contrib/rules_oci/issues/183

**Comment by prestonvanloon** [comment]
Another blocker for CI. https://github.com/bazel-contrib/rules_oci/issues/292

**Comment by prestonvanloon** [comment]
Still blocked by https://github.com/bazel-contrib/rules_oci/issues/292

**Comment by prestonvanloon** [comment]
With https://github.com/prysmaticlabs/prysm/pull/12428/commits/e948da3e3d28bf97ab94037eddf1786868815d00, I think we can say we are unblocked and merge. 

If build times increase significantly then we may have to revert this PR. The flag removed only matters when fetching dependencies where it would hit an internal endpoint before reaching the external network. In theory, it shouldn't be much different but possibly an issue. 

I propose that we review this PR and merge targeting `gcr.io/prylabs-dev/prysm` for one release cycle, test further, then remove the old rules_docker and use the canonical gcr.io image URLs / names.

File: .buildkite-bazelrc [hunk 1]
```diff
@@ -12,7 +12,8 @@
 #build:remote-cache --disk_cache=
 build:remote-cache --remote_download_toplevel
 build:remote-cache --remote_cache=grpc://bazel-remote-cache:9092
-build:remote-cache --experimental_remote_downloader=grpc://bazel-remote-cache:9092
+# Does not work with rules_oci.
+#build:remote-cache --experimental_remote_downloader=grpc://bazel-remote-cache:9092
```
Comment (james-prysm) [inline .buildkite-bazelrc:16]: this is now commented out right? what does this do?

File: tools/BUILD.bazel [hunk 2]
```diff
@@ -99,3 +100,26 @@ container_image(
     user = "root",
     visibility = ["//visibility:public"],
 )
+
+# bash tar layer for docker images.
+genrule(
+    name = "bash_tar",
```
Comment (james-prysm) [inline tools/BUILD.bazel:106]: why is bash_tar and libtinfo6_tar needed, does one handle something differently than the other?
perhaps related to comment # bash tar layer for docker images.

File: tools/multi_arch.bzl [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    ]
+
+multiarch_transition = transition(
+    implementation = _multiarch_transition,
+    inputs = [],
+    outputs = ["//command_line_option:platforms"],
+)
+
+def _impl(ctx):
+    return DefaultInfo(files = depset(ctx.files.image))
```
Comment (james-prysm) [inline tools/multi_arch.bzl:16]: what does this do?

File: .buildkite-bazelrc (hunk 1 above)
Comment (prestonvanloon) [reply .buildkite-bazelrc:16]: Yeah, it doesn't work well with rules_oci. I'll amend the comment above to link to the issue. Basically, the method of pulling base images is non-hermetic and is causing issues in CI (i think). No confident in the root cause this flag is problematic with rules_oci.

https://github.com/bazel-contrib/rules_oci/issues/292

Comment (prestonvanloon) [reply .buildkite-bazelrc:16]: Add ref to the above issue in the commentary.

File: tools/BUILD.bazel (hunk 2 above)
Comment (prestonvanloon) [reply tools/BUILD.bazel:107]: Added commentary to explain this

File: tools/multi_arch.bzl (hunk 3 above)
Comment (prestonvanloon) [reply tools/multi_arch.bzl:18]: It's a "simple" starlark transition implementation. All rules have must return a provider of some kind. This one returns the default provider with information about what should be built in order to fulfill this rule.

**Review by james-prysm — APPROVED** [review]
Thanks for the explanations

**Pull request merged**